website/project/views/comment.py

```diff
@@ -17,30 +18,69 @@
 
 
 @file_updated.connect
-def update_comment_root_target_file(self, node, event_type, payload, user=None):
-    if event_type == 'addon_file_moved':
+def update_file_guid_referent(self, node, event_type, payload, user=None):
+    if event_type == 'addon_file_moved' or event_type == 'addon_file_renamed':
         source = payload['source']
         destination = payload['destination']
         source_node = Node.load(source['node']['_id'])
         destination_node = node
-
-        if (source.get('provider') == destination.get('provider') == 'osfstorage') and source_node._id != destination_node._id:
-            obj = FileNode.load(source.get('path').strip('/'))
-            update_folder_contents([obj], source_node, destination_node)
-
-
-def update_folder_contents(children, source_node, destination_node):
+        file_guids = FileNode.resolve_class(source['provider'], FileNode.ANY).get_file_guids(
+            materialized_path=source['materialized'] if source['provider'] != 'osfstorage' else source['path'],
+            provider=source['provider'],
+            guids=[],
+            node=source_node)
+
+        if event_type == 'addon_file_renamed' and (source['provider'] == 'osfstorage' or source['provider'] == 'box'):
+            return
+        if event_type == 'addon_file_moved' and (source['provider'] == destination['provider'] and (source['provider'] == 'osfstorage' or source['provider'] == 'box')) and source_node == destination_node:
+            return
+
+        for guid in file_guids:
+            obj = Guid.load(guid)
+            if source_node != destination_node and Comment.find(Q('root_target', 'eq', guid)).count() != 0:
+                update_comment_node(guid, source_node, destination_node)
+
+            if not (source['provider'] == destination['provider'] and source['provider'] == 'osfstorage'):
```

Comment:
Not a big deal, but this might be a bit easier to follow if it were

```py
if source['provider'] != destination['provider'] or source['provider'] != 'osfstorage':
```